agent/engine/common_integ_test.go

```diff
@@ -117,14 +117,18 @@ func verifyContainerRunningStateChangeWithRuntimeID(t *testing.T, taskEngine Tas
 		"Expected container runtimeID should not empty")
 }
 
-func verifyExecAgentRunningStateChange(t *testing.T, taskEngine TaskEngine) {
+func verifyExecAgentStateChange(t *testing.T, taskEngine TaskEngine,
+	expectedStatus apicontainerstatus.ManagedAgentStatus, waitDone chan<- struct{}) {
 	stateChangeEvents := taskEngine.StateChangeEvents()
-	event := <-stateChangeEvents
-	containerEvent := event.(api.ContainerStateChange)
-	assert.NotEmpty(t, containerEvent.ManagedAgents, "Expected exec-agent event has no managed agents")
-	if containerEvent.ManagedAgents != nil {
-		assert.Equal(t, apicontainerstatus.ManagedAgentRunning, containerEvent.ManagedAgents[0].Status,
-			"expected managedAgent container state change event did not match actual event")
+	for event := range stateChangeEvents {
+		if containerEvent, ok := event.(api.ContainerStateChange); ok {
```

Comment:
there's a possibility here that we receive an `api.ContainerStateChange` but not the one we were looking for. Is there a way to make sure it's a container event for managed agent? 

I am thinking actually that maybe `api.ContainerStateChange` was meant for actual changes in the container state, maybe we should consider to create a `api.ManagedAgentStateChange`? Not for this PR maybe, but something to consider

Reply:
Even with the redundant event filter I was seeing events with NONE managedAgent status prior to the expected events, and this function sifted past these events.  
I agree that we might want to differentiate the events meant to converse with Docker from the events meant to converse with ECS Control Plane.  Both could be considered ContainerEvents.
(not in present pr though.)